File: src/Sail.Core/Entities/Certificate.cs
namespace Sail.Core.Entities;

public class Certificate
{
    public Guid Id { get; set; }
    public string? Cert { get; init; }
    public string? Key { get; init; }
    public List<SNI>? SNIs { get; init; }
    public DateTimeOffset CreatedAt { get; init; } = DateTimeOffset.UtcNow;
    public DateTimeOffset UpdatedAt { get; init; } = DateTimeOffset.UtcNow;
}

File: web/src/types/certificate.ts
export interface SNI {
  id: string;
  name: string;
  hostName: string;
  createdAt: string;
  updatedAt: string;
}

export interface Certificate {
  id: string;
  name?: string;
  cert?: string;
  key?: string;
  snIs?: SNI[];
  createdAt: string;
  updatedAt: string;
}

